sdk/communication/AzureCommunicationChat/Source/Registrar/RegistrarClient.swift

```diff
@@ -80,79 +99,113 @@ internal class RegistrarClient {
         return try? JSONEncoder().encode(data)
     }
 
-    /// Sets the authentication header on a Registrar request.
+    /// Create a Registrar POST request.
     /// - Parameters:
-    ///   - request: The  HTTP request.
-    ///   - completionHandler: Returns the request, or an error if the request failed to be authenticated.
-    private func setAuthHeader(
-        on request: URLRequest,
-        completionHandler: @escaping (URLRequest?, AzureError?) -> Void
+    ///   - clientDescription: RegistrarClientDescription.
+    ///   - transports: RegistrarTranports, a transport contains the APNS token as the path.
+    ///   - completionHandler: Returns the POST request.
+    private func createPostRequest(
+        with clientDescription: RegistrarClientDescription,
+        for transports: [RegistrarTransport],
+        completionHandler: @escaping (HTTPRequest?, AzureError?) -> Void
```

Comment:
This looks strange. Only the PipelineClient initializer (in this case `RegistrarClient`) shoudl accept a transport stage as a parameter.

Reply:
I think this is what we supposed it to be. We have two "transport" related objects: (A) the **RegistrarTransport** which contains the APNS token as the path, (B) the **TransportStage** which is required to initialize PipelineClient.  In this "createPostRequest()", we pass (A) as the argument. And (B) has been passed as a parameter in super.init() inside RegistrarClient's initializer.

Reply:
Yes, this is what was confusing to me. As mentioned elsewhere, `RegistrarTransport` should be renamed since it does not send any network traffic.

Another question: why is it an array and not just an object?

Reply:
Why is it an array but not an object? It is designed to be in this way. Based on Registrar Service's document, the "transports" in request body should be a [string: list<RegistrarTransport>] dictionary. So we pass a collection of RegistrarTransport but not a single RegistrarTransport object.

Reply:
About renaming:not sure if it is good to rename it. We name it as "transport" since this is what is named in the Registrar Service's API document. Also I check Android side and they also name it as "transport".

Reply:
Neither of those are compelling reasons to call it XTransport if it causes confusion with existing concepts in the iOS SDK. This object does not do any network calls, so it is not a transport layer. You can call it `RegistrarTransportSettings` or something like that.

Reply:
Thanks. Already converted the name to "RegistrarTransportSettings". Will push the change later.